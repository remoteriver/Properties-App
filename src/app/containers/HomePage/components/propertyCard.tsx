import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import { BrandingColor, Property } from 'types/Property';
import { Grid } from '@material-ui/core';

export interface Props {
  property: Property;
  children: JSX.Element;
}

const useStyles = makeStyles({
  root: { position: 'relative' },
  card: {
    position: 'relative',
  },
  cardDisabled: {
    position: 'relative',
    opacity: 0.5,
  },
  media: {
    height: 240,
  },
  banner: {
    backgroundColor: (props: BrandingColor) => props.primary,
  },
  children: {
    position: 'absolute',
    right: 15,
    bottom: 22,
  },
});

export const PropertyCard = (props: Props) => {
  const { property, children } = props;
  const classes = useStyles(property.agency.brandingColors);

  const [hover, setHover] = useState(false);

  const handleMouseEnter = () => {
    setHover(true);
  };
  const handleMouseLeave = () => {
    setHover(false);
  };

  return (
    <div className={classes.root}>
      <Card
        className={
          property?.disabled === true ? classes.cardDisabled : classes.card
        }
        onMouseEnter={() => handleMouseEnter()}
        onMouseLeave={() => handleMouseLeave()}
      >
        <CardContent className={classes.banner}>
          <Grid container justifyContent="space-between" alignItems="center">
            <Grid item>
              <img src={property.agency.logo} alt="agency logo" />
            </Grid>
          </Grid>
        </CardContent>
        <CardMedia
          className={classes.media}
          image={property.mainImage}
          title="main image"
        />
        <CardContent>
          <Grid container justifyContent="space-between" alignItems="center">
            <Grid item>
              <Typography gutterBottom variant="h5" component="span">
                {property.price}
              </Typography>
            </Grid>
            <Grid item hidden={!hover}>
              <div className={classes.children}>{children}</div>
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    </div>
  );
};
