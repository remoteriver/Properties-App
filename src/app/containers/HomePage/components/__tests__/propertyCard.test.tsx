import { Button } from '@material-ui/core';
import * as React from 'react';
import { Provider, useDispatch, useSelector } from 'react-redux';
import { createRenderer } from 'react-test-renderer/shallow';
import { configureAppStore } from 'store/configureStore';
import { PropertyCard } from '../propertyCard';

const renderer = createRenderer();

describe('<HomePage />', () => {
  const mockProps = {
    property: {
      id: 1,
      price: '$726,500',
      agency: {
        brandingColors: {
          primary: '#ffe512',
        },
        logo:
          'https://i1.au.reastatic.net/170x32/d9e65c666e427e655fb63dcfe73f50d4ac7ff9a58c173db9474bd92e75b01070/main.gif',
      },
      mainImage:
        'https://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg',
    },
  };

  beforeAll(() => {});

  it('should render and match the snapshot', () => {
    //act
    renderer.render(
      <PropertyCard {...mockProps}>
        <Button>Add Property</Button>
      </PropertyCard>,
    );
    const renderedOutput = renderer.getRenderOutput();

    //assert
    expect(renderedOutput).toMatchSnapshot();
  });
});
