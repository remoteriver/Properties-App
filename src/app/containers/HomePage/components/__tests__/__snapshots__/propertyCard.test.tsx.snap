// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-root-1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-banner-3 makeStyles-banner-5"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justifyContent="space-between"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <img
            alt="agency logo"
            src="https://i1.au.reastatic.net/170x32/d9e65c666e427e655fb63dcfe73f50d4ac7ff9a58c173db9474bd92e75b01070/main.gif"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-2"
      image="https://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
      title="main image"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justifyContent="space-between"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Typography))
            component="span"
            gutterBottom={true}
            variant="h5"
          >
            $726,500
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          hidden={true}
          item={true}
        >
          <div
            className="makeStyles-children-4"
          >
            <WithStyles(ForwardRef(Button))>
              Add Property
            </WithStyles(ForwardRef(Button))>
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</div>
`;
